src/main/java/carbon/task/Event.java

```diff
@@ -0,0 +1,133 @@
+package carbon.task;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.Temporal;
+
+import carbon.error.CarbonException;
+import carbon.error.InvalidFlagException;
+import carbon.error.InvalidParamException;
+import carbon.error.InvalidTimeException;
+
+/**
+ * Encapsulates information regarding Events.
+ */
+public class Event extends Task {
+    public static final String FLAG = " /at";
+    private static final int TYPEKEY = Task.getTypeKey(Task.Type.EVENT);
+
+    private Temporal dateTime;
+
+    private Event(String name, Boolean isDone, Temporal dateTime) {
+        super(name, isDone);
+        this.dateTime = dateTime;
+    }
+
+    private static String extractName(String input) {
```

Comment:
Maybe it's not very clear in code, but extractName uses the `String::substring` method to get the part of the user input after "event " until the flag (" /at"), essentially capturing the name that the user assigns to the task. 

I understand the point on checking validity though (same as the marking methods for `TaskList`). I currently have most of the validation decentralised among classes, so the `Parser` class mainly handles just the parsing and calling of processing methods. 

I do think having `Parser` perform validation makes sense too, though I'm not sure if it will result in overlapping input processing.